packages/medusa/src/services/order.js

```diff
@@ -583,9 +595,17 @@ class OrderService extends BaseService {
         )
       }
 
+      const evaluatedNoNotification =
+        no_notification !== undefined
+          ? no_notification
+          : shipment.no_notification
+
       const shipmentRes = await this.fulfillmentService_
         .withTransaction(manager)
-        .createShipment(fulfillmentId, trackingLinks, metadata)
+        .createShipment(fulfillmentId, trackingLinks, {
+          metadata,
+          no_notification: evaluatedNoNotification,
+        })
```

Comment:
The current method signature in FulfillmentService (`/src/services/fulfillment.js line:253`): `async createShipment(fulfillmentId, trackingLinks, metadata)`
Update to take `config` as last variable instead of `metadata`.

Also make sure that `fufillment.no_notification` is updated when calling `createShipment` with `no_notification`